homeassistant/components/onvif/__init__.py

```diff
@@ -61,9 +62,19 @@ async def async_setup(hass: HomeAssistant, config: dict):
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Set up ONVIF from a config entry."""
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {}
+
     if not entry.options:
         await async_populate_options(hass, entry)
 
+    device = ONVIFDevice(hass, entry)
+
+    if not await device.async_setup():
+        return False
```

Comment:
Should this `raise ConfigEntryNotReady` so Home Assistant will retry ?

Reply:
a `False` return, no - user action will be required to recover.  I will, however raise if `not device.available` after setup is complete.